### PR TITLE
PHP 8.1 runtimes

### PR DESCRIPTION
**Pull request by mykiwi**
This pull request adds [PHP 8.1](https://www.php.net/archive/2021.php#2021-08-19-1) runtimes (PHP 8.1.0beta3)
Heavily inspired from #678

### To test it:

Name | Version | Compatible runtimes | Version ARN
-- | -- | -- | --
php-81 | 3 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81:3
php-81-fpm | 3 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81-fpm:3

### CODE REVIEW

**Comment by deleugpn** [comment]
Thanks! We'll likely combine this with our plan on automating the build next week. Did you test if this build is working as expected?

**Comment by mykiwi** [comment]
Not yet, it is currently in the build process

**Comment by mykiwi** [comment]
When I run `php runtime/layers/tests.php` this does not works, but also on older images too :thinking: 
Example of what it does with an older image:
```
$ docker run --rm --entrypoint pwd bref/php-80-fpm
docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:385: applying cgroup configuration for process caused: error while starting unit "docker-64bc0017cb977edbda42ec78b30267ac5258c278ff15baf1912c37090d9de2a5.scope" with properties [{Name:Description Value:"libcontainer container 64bc0017cb977edbda42ec78b30267ac5258c278ff15baf1912c37090d9de2a5"} {Name:Slice Value:"system.slice"} {Name:PIDs Value:@au [775108]} {Name:Delegate Value:true} {Name:MemoryAccounting Value:true} {Name:CPUAccounting Value:true} {Name:IOAccounting Value:true} {Name:TasksAccounting Value:true} {Name:DefaultDependencies Value:false}]: Failed to activate service 'org.freedesktop.systemd1': timed out (service_start_timeout=25000ms): unknown.
```

How can I ensure this works correctly?

**Comment by mnapoli** [comment]
OK we went over these changes, but we're afraid of a long feedback loop of:

1. we merge these changes
2. we release new layers
3. someone tests the layers and figures out something's not working
4. new PR with the changes
5. back to step 1

Such a feedback loop is very long and takes a lot of time. What would be best instead, before merging, is knowing that the layers do indeed work.

One way to do that is to build the layers locally and publish them to your own AWS account (`make runtimes`). Feel free to disable some regions (e.g. only publish to `us-east-1`), that will be faster to publish and simpler to cleanup after.

Then share the layers ARN here, so that people can test it out. Once we are confident the layers are working, we can merge them and publish them.

**Comment by mykiwi** [comment]
:ok_hand:  

> Then share the layers ARN here, so that people can test it out.

<span><span data-analytics="refreshEnabledV2" data-analytics-type="variant">

Name | Version | Compatible runtimes | Version ARN
-- | -- | -- | --
php-81 | 1 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81:1
php-81-fpm | 1 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81-fpm:1

</span></span>

**Comment by mnapoli** [comment]
Thanks! I shared on twitter too (I won't have time to test today personally): https://twitter.com/brefphp/status/1422185768978681859

**Comment by mykiwi** [comment]
PHP 8.1.0beta2

<span><span data-analytics="refreshEnabledV2" data-analytics-type="variant">

Name | Version | Compatible runtimes | Version ARN
-- | -- | -- | --
php-81 | 2 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81:2
php-81-fpm | 2 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81-fpm:2

</span></span>

**Comment by mykiwi** [comment]
PHP 8.1.0beta3

Name | Version | Compatible runtimes | Version ARN
-- | -- | -- | --
php-81 | 3 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81:3
php-81-fpm | 3 | provided.al2 | arn:aws:lambda:us-east-1:670541007614:layer:php-81-fpm:3

**Comment by shouze** [comment]
@mykiwi I'm ok to test it but prior to test lambda layers, I would prefer to make that gradually against our apps:
1. Test locally on our docker containers, but for that I would prefer of course to pull bref preview containers for php 8.1
2. Then test on AWS with the preview published layers.

Do you think it's possible to publish at least `bref/php-81-fpm-dev`? (BTW I've seen that this is the one you have commented at the moment ^^).

Also, as I use some extra php extensions (in fact just one: redis) in many of our apps do you think it's also possible to publish the `bref/build-php-81` docker image too?

And @Nyholm... this would imply then to build and publish first docker preview images & layers too :angel:

**Comment by Nyholm** [comment]
As soon as `bref/build-php-81` is on dockerhub, we can start publishing all/most extensions.

**Comment by mnapoli** [comment]
👍  cool!

Just to be transparent, the current status of this PR is basically waiting for someone telling us that these layers work 😄

**Comment by mnapoli** [comment]
Actually now that we have the build pipeline fully automated, let's give this a try!

Thank you @mykiwi!

File: runtime/Makefile [hunk 1]
```diff
@@ -37,26 +37,32 @@ docker-images:
...
 	# - console
 	cd layers/console ; docker build -t bref/php-73-console --build-arg PHP_VERSION=73 .
 	cd layers/console ; docker build -t bref/php-74-console --build-arg PHP_VERSION=74 .
 	cd layers/console ; docker build -t bref/php-80-console --build-arg PHP_VERSION=80 .
+	cd layers/console ; docker build -t bref/php-81-console --build-arg PHP_VERSION=81 .
 	# Other Docker images
 	cd layers/fpm-dev ; docker build -t bref/php-73-fpm-dev --build-arg PHP_VERSION=73 .
 	cd layers/fpm-dev ; docker build -t bref/php-74-fpm-dev --build-arg PHP_VERSION=74 .
 	cd layers/fpm-dev ; docker build -t bref/php-80-fpm-dev --build-arg PHP_VERSION=80 .
+	#cd layers/fpm-dev ; docker build -t bref/php-81-fpm-dev --build-arg PHP_VERSION=81 . # xdebug not ok
```
Comment (shouze) [inline runtime/Makefile:65]: Would have been nice to enabling it but without xdebug ATM, WDYT? I can try a PR if :ok:

Comment (mnapoli) [reply runtime/Makefile:65]: 👍  sounds good to me

Comment (mykiwi) [reply runtime/Makefile:65]: I will try to rework on it if my free time allows me, go :+1: for trying on your side

Comment (deleugpn) [reply runtime/Makefile:65]: Whoever picks this up, consider taking a quick look at https://github.com/brefphp/bref/blob/master/Makefile#L18 to see if it's easy to start publishing php-81 images on Docker Hub, that would help me test PHP 8.1!

Comment (shouze) [reply runtime/Makefile:65]: @deleugpn @mykiwi @mnapoli the PR is here :wink: 
https://github.com/brefphp/bref/pull/1007

@Nyholm I also have started some changes in extra extensions repository so will make a PR once php-81 docker images are published on docker hub.

Comment (shouze) [reply runtime/Makefile:65]: Sorry #1008 as an upfix of #1007... but I've also prepared the extra extensions PR https://github.com/brefphp/extra-php-extensions/pull/261

**Pull request merged**